client/core/core.go

```diff
@@ -6670,3 +6672,275 @@ func (c *Core) WalletLogFilePath(assetID uint32) (string, error) {
+			return fmt.Errorf("unable to get quote unit info for %v: %v", cord.QuoteSymbol, err)
+		}
+		ordReader := &OrderReader{
+			Order:         cord,
+			BaseUnitInfo:  baseUnitInfo,
+			QuoteUnitInfo: quoteUnitInfo,
+		}
+
+		timestamp := encode.UnixTimeMilli(int64(cord.Stamp)).Local().Format(time.RFC3339Nano)
+		err = csvWriter.Write([]string{
+			cord.Host,                     // Host
+			ord.Order.ID().String(),       // Order ID
+			cord.BaseSymbol,               // Base
+			cord.QuoteSymbol,              // Quote
+			ordReader.BaseQtyString(),     // Base Quantity
+			ordReader.SimpleRateString(),  // Order Rate
+			ordReader.AverageRateString(), // Actual Rate
+			ordReader.BaseAssetFees(),     // Base Fees
+			ordReader.QuoteAssetFees(),    // Quote Fees
+			ordReader.Type.String(),       // Type
+			ordReader.SideString(),        // Side
+			cord.TimeInForce.String(),     // Time in Force
+			ordReader.StatusString(),      // Status
+			cord.TargetOrderID.String(),   // Target Order ID
+			ordReader.FilledPercent(),     // Filled
+			ordReader.SettledPercent(),    // Settled
+			timestamp,                     // Time
+		})
+		if err != nil {
+			return fmt.Errorf("error writing orders CSV: %v", err)
+		}
+		csvWriter.Flush()
+		err = csvWriter.Error()
+		if err != nil {
+			return fmt.Errorf("error writing orders CSV: %v", err)
+		}
+		return nil
+	}, ordersFile.Close, nil
+}
+
+func deleteMatchFn(matchesFileStr string) (perMatchFn func(*db.MetaMatch, bool) error, cleanUpFn func() error, err error) {
+	matchesFile, err := createFile(matchesFileStr)
+	if err != nil {
+		return nil, nil, fmt.Errorf("problem opening orders file: %v", err)
+	}
+	csvWriter := csv.NewWriter(matchesFile)
+	csvWriter.UseCRLF = runtime.GOOS == "windows"
+
+	err = csvWriter.Write([]string{
+		"Host",
+		"Base",
+		"Quote",
+		"Match ID",
+		"Order ID",
+		"Quantity",
+		"Rate",
+		"Swap Fee Rate",
+		"Swap Address",
+		"Status",
+		"Side",
+		"Secret Hash",
+		"Secret",
+		"Contract Data",
+		"Counter Contract",
+		"Counter TxData",
+		"Maker Swap Coin ID",
+		"Maker Redeem Coin ID",
+		"Taker Swap Coin ID",
+		"Taker Redeem Coin ID",
+		"Refund Coin ID",
+		"Time",
+	})
+	if err != nil {
+		matchesFile.Close()
+		return nil, nil, fmt.Errorf("error writing matches CSV: %v", err)
+	}
+	csvWriter.Flush()
+	err = csvWriter.Error()
+	if err != nil {
+		matchesFile.Close()
+		return nil, nil, fmt.Errorf("error writing matches CSV: %v", err)
+	}
+	return func(mtch *db.MetaMatch, isSell bool) error {
+		numToStr := func(n interface{}) string {
+			return fmt.Sprintf("%d", n)
+		}
+		base, quote := mtch.MetaData.Base, mtch.MetaData.Quote
+		baseSymbol, quoteSymbol := dex.BipIDSymbol(base), dex.BipIDSymbol(quote)
+
+		// If we are maker and buying, we are buying base with quote.
+		makerBuyingAsset, makerSellingAsset := base, quote
+		// If we are either not maker or not buying, invert it. Double
+		// inverse would be no change.
+		if (mtch.Side == order.Taker) != isSell {
```

Comment:
Tricky logic.  To check, if I write this out more verbosely:

- maker + selling = makerSwap is base asset (takerSwap is quote)
- maker + buying = makerSwap is quote asset (takerSwap is base)
- taker + selling = makerSwap is quote asset (takerSwap is base)
- taker + buying = makerSwap is base asset (takerSwap is quote)

```go
func matchAssets(base, quote uint32, sell, taker bool) (makerSwapAsset, takerSwapAsset uint32) {
	if sell {
		if taker {
			// maker buy, taker sell
			takerSwapAsset = base // me
			makerSwapAsset = quote
		} else {
			// maker sell, taker buy
			takerSwapAsset = quote
			makerSwapAsset = base // me
		}
	} else {
		if taker {
			// maker sell, taker buy
			takerSwapAsset = quote // me
			makerSwapAsset = base
		} else {
			// maker buy, taker sell
			takerSwapAsset = base
			makerSwapAsset = quote // me
		}
	}
	return
}
```

I'm having a hard time reasoning out if these are equal, partly because of the different variable meanings, where `makerBuyingAsset` refers to the maker's redeem and the taker's init.

Going to make some test cases.

Reply:
I got really bogged down today and didn't actually get to verifying this, but it seems to be the same.

Reply:
https://go.dev/play/p/3oLXnYUtWhQ

I can change the names to takerSwapAsset and makerSwapAsset

Reply:
Changed the name.

Reply:
Ty for the verification.